src/profile-logic/profile-data.js

```diff
@@ -1522,3 +1522,111 @@ export function getTreeOrderComparator(
     return compareCallNodes(callNodeA, callNodeB);
   };
 }
+
+/**
+ * This function walks up the from a clicked call node, and tries to find the best
+ * call node to select. This is the root-most call node that is drawn on the same
+ * category path as the clicked call node.
```

Comment:
I'd try to avoid the term "category path" in this description; at first I thought this referred to the canvas path that is filled for a category fill. But no, it's more like `callNodePath.map(categoryForCallNode)`. How about:

"This is the root-most call node for which, if selected, only the clicked category is highlighted in the thread activity graph. In other words, it's the root-most call node which only 'contains' samples whose sample category is the clicked category."